share/completions/jj.fish

```diff
@@ -1 +1,10 @@
-jj util completion fish | source
+# The reason for `__this-command-does-not-exist` is that, if dynamic completion
+# is not implemented, we'd like to get an error reliably. However, the
+# behavior of `jj` without arguments depends on the value of a config, see
+# https://jj-vcs.github.io/jj/latest/config/#default-command
+if set -l completion (COMPLETE=fish jj __this-command-does-not-exist 2>/dev/null)
+    # jj is new enough for dynamic completions to be implemented
+    echo "$completion" | source
```

Comment:
This joins multiple lines by spaces which may break if future `jj` outputs multiple lines.
To reduce surprise and eyebrow-raising, we should use something like `printf %s\n $completion`.
(could also elide the variable with `if ...; end | source`)

Reply:
I think I'll go with `printf`. I think that, in practice, `if |source` would work, but I'm not entirely sure whether some of the old versions of fish might print the error from `jj this-command-doesn-not-exist` to stdout.